src/components/common/Input/index.js

```diff
@@ -1,22 +1,38 @@
 import React from 'react';
 import { string, bool, oneOfType, array } from 'prop-types';
 import { View, TextInput, Text } from 'react-native';
+
+import common from 'constants/commonStyles';
 import styles from './styles';
 
-const Input = ({ label, invalid, error, ...props }) => (
+const Input = ({ label, invalid, error, showErrorMessage, short, ...props }) => (
   <>
     {label && <Text style={styles.label}>{label}</Text>}
-    <View>
-      <TextInput style={[styles.input, invalid && styles.inputBorderError]} {...props} />
-      {error && <Text>{error}</Text>}
+    <View style={styles.container}>
+      <TextInput
+        style={[styles.input, invalid && styles.inputBorderError, short && common.shortInputWidth]}
+        {...props}
+      />
+      {showErrorMessage && (
+        <View style={styles.errorContainer}>
+          {error && <Text style={styles.error}>{error}</Text>}
+        </View>
+      )}
     </View>
   </>
 );
 
 Input.propTypes = {
   label: string,
-  error: string,
+  error: array,
   invalid: oneOfType([bool, string, array]),
+  showErrorMessage: bool,
+  short: bool,
+};
+
+Input.defaultProps = {
+  showErrorMessage: false,
+  short: false,
```

Comment:
I see you use short in all the cases for the sign-up, maybe you could change true as the default one